src/agent.ts

```diff
@@ -254,26 +248,26 @@ export class Agent {
   private async _askUserForConfirmation(
     functionCall: OpenAIFunctionCall
   ): Promise<boolean> {
-
-    const functionCallStr =
-      `\`\`\`\n${functionCall.name} (${functionCall.arguments})\n\`\`\`\n`;
-
     if (this._autoPilotMode) {
+      const functionCallStr =
+        `\`\`\`\n${functionCall.name} (${functionCall.arguments})\n\`\`\`\n`;
```

Comment:
If the same variable is used by both conditional branches it should be hoisted above the if condition. This way it's accessible to both branches without re-declaring it